File: LeThinhAPI/Event.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace LeThinhAPI.Models
{
    public class Event
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Title { get; set; } = string.Empty;

        public string? Description { get; set; }

        [Required]
        public DateTime StartTime { get; set; }

        [Required]
        public DateTime EndTime { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? DeletedAt { get; set; }

        [Required]
        public int UserId { get; set; }

        [ForeignKey("UserId")]
        [JsonIgnore]
        public virtual User CreatedBy { get; set; } = null!;

        [JsonIgnore]
        public virtual ICollection<Reminder> Reminders { get; set; } = new List<Reminder>();

        [JsonIgnore]
        public virtual ICollection<EventAssignment> Assignments { get; set; } = new List<EventAssignment>();
    }
}

File: lib/models/event.dart
class Event {
  final int id;
  final String title;
  final String? description;
  final DateTime startTime;
  final DateTime endTime;

  Event({
    required this.id,
    required this.title,
    this.description,
    required this.startTime,
    required this.endTime,
  });

  // Factory constructor để tạo một Event từ JSON
  // Dữ liệu JSON này được gửi về từ backend .NET của bạn
  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'] ?? 0,
      title: json['title'] ?? 'Không có tiêu đề',
      description: json['description'],
      // Chuyển đổi chuỗi String ISO 8601 từ backend thành đối tượng DateTime của Dart
      startTime: DateTime.parse(json['startTime'] ?? ''),
      endTime: DateTime.parse(json['endTime'] ?? ''),
    );
  }
}
